### ISSUE TITLE
Navigate filters sometime throw errors when navigating to builds

### ISSUE THREAD

**Issue by benjaminapetersen** [openshift/origin-web-console#1942]
`navigateResourceURL` and `navigateEventInvolvedObjectURL` (and perhaps the others) occasionally cause an error to be thrown when navigating to builds:

![screen shot 2017-08-16 at 11 47 41 am](https://user-images.githubusercontent.com/280512/29372357-e495c9dc-8278-11e7-99ad-1bdb11c7ca63.png)

error on line 105:

![screen shot 2017-08-16 at 11 49 50 am](https://user-images.githubusercontent.com/280512/29372416-08981d26-8279-11e7-9d15-4ac6c533f97e.png)

The breadcrumbs are not generated correctly due to the URL having a slightly different format.  
Breadcrumb when error thrown:

![screen shot 2017-08-16 at 11 51 24 am](https://user-images.githubusercontent.com/280512/29372481-45356216-8279-11e7-8470-ea4954070c06.png)

Breadcrumb when error not thrown:
![screen shot 2017-08-16 at 11 51 31 am](https://user-images.githubusercontent.com/280512/29372487-490c3ea0-8279-11e7-9046-a2764b6d79d9.png)

URL generating breadcrumbs with error:

`https://localhost:9000/dev-console/project/myproject/browse/builds-noconfig/mmmmmm-cake-1?tab=details`

URL generating correct breadcrumbs:

`https://localhost:9000/dev-console/project/myproject/browse/builds/mmmmmm-cake/mmmmmm-cake-1?tab=details`

So problem seems to be that the `navigate*` functions may think a build has no build config.

@spadgett

**Comment by benjaminapetersen** [comment]
This can be reproduced on the Monitoring page and in the Notification drawer.

Resolution per conversation is probably to look @ the `Build Config` annotation on the build rather than check the URL (via new `ownerRefs`).

**Comment by spadgett** [comment]
We might just remove the build config name from the URL like we did for replication controllers owned by a deployment config. Then you can build the URL from just the kind and name without needing other data. If we do that, we'd want a redirect for the old URL for old links.

We probably have a problem for pipeline builds vs regular builds as well since we use a `pipeline` URL segment for pipeline builds.

**Comment by benjaminapetersen** [comment]
Hmm. Yeah, this one does seem complex then, since we have so many routes using `BuildController`:

```javascript 
.when('/project/:project/browse/builds/:buildconfig/:build', {
  controller: 'BuildController',
  //
})
.when('/project/:project/browse/pipelines/:buildconfig/:build', {
  controller: 'BuildController',
  //
})
.when('/project/:project/browse/builds-noconfig/:build', {
  controller: 'BuildController',
  //
})
.when('/project/:project/browse/pipelines-noconfig/:build', {
  controller: 'BuildController',
  //
})
```

**Comment by jwforres** [comment]
The javascript error is being fixed in #2403 
We will make the longer term URL change next release
